libr/core/io.c

```diff
@@ -97,23 +97,42 @@ R_API int r_core_dump(RCore *core, const char *file, ut64 addr, ut64 size, int a
 
 R_API int r_core_write_op(RCore *core, const char *arg, char op) {
 	int i, j, len, ret = false;
-	char *str;
+	char *str = NULL;
 	ut8 *buf;
 
 	// XXX we can work with config.block instead of dupping it
 	buf = (ut8 *)malloc (core->blocksize);
-	str = (char *)malloc (strlen (arg)+1);
-	if (buf == NULL || str == NULL)
+	if (!buf)
 		goto beach;
 	memcpy (buf, core->block, core->blocksize);
-	if (op!='e') {
-		len = r_hex_str2bin (arg, (ut8 *)str);
-		if (len < 0) {
-			eprintf ("Invalid hexpair string\n");
+
+	// fill key buffer either from arg or from clipboard
+	if (arg) {  // parse arg for key
+		str = (char *)malloc (strlen (arg)+1);
```

Comment:
> better use calloc to avoid uninitialized memory in case r_hex_str2bin doesnt fills up all the buffer

After giving it a closer look, zero-filling the buffer by using ```calloc()``` won't change anything but make it slower. What you're suggesting makes sense for string operations that might end up non-zero-terminated, but ```r_hex_str2bin()``` writes raw bytes (including zeroes) and returns the number of bytes it has written to the buffer. Buffer length is chosen such that it is *guaranteed to be much too large*, and all the code that works on ```str``` thereafter relies on ```len``` for termination.

If there's anything missing from this code, it is asserting that ```len <= strlen (arg) + 1``` to detect stack corruption, but given that *two nibble chars input result in one byte output*, one can safely omit that.

Btw, the ```+ 1``` turns out to be unnecessary. ```malloc(0)``` is fine to ```free()```, and ```len == 0``` will be caught right after.